packages/@o3r/design/src/core/design-token/parsers/design-token.parser.ts

```diff
@@ -0,0 +1,172 @@
+import type {
+  DesignToken,
+  DesignTokenExtensions,
+  DesignTokenGroup,
+  DesignTokenGroupExtensions,
+  DesignTokenNode,
+  DesignTokenSpecification
+} from '../design-token-specification.interface';
+import {
+  DesignTokenTypeStrokeStyleValue,
+  isDesignToken,
+  isDesignTokenGroup,
+  isTokenTypeStrokeStyleValueComplex
+} from '../design-token-specification.interface';
+
+interface ParentReference {
+  name: string;
+  tokenNode: DesignTokenGroup;
+}
+
+const tokenReferenceRegExp = /\{([^}]+)\}/g;
+
+const getTokenReferenceName = (tokenName: string, parents: string[]) => (`${parents.join('.')}.${tokenName}`);
+const getCssVarName = (tokenName: string, parents: string[]) => (getTokenReferenceName(tokenName, parents).replace(/[ .]+/g, '-'));
+const getGroups = (tokenName: string, parents: string[]) => (getTokenReferenceName(tokenName, parents).split(/[ .]+/).slice(0, -1));
+const getExtensions = (parentNode: DesignTokenNode[]) => parentNode.reduce((acc, node) => ({...acc, ...node.$extensions}), {} as DesignTokenGroupExtensions & DesignTokenExtensions);
+const getReferences = (cssRawValue: string) => Array.from(cssRawValue.matchAll(tokenReferenceRegExp)).map(([,tokenRef]) => tokenRef);
+
+// eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
+const renderCssTypeStrokeStyleValue = (value: DesignTokenTypeStrokeStyleValue | string) => isTokenTypeStrokeStyleValueComplex(value) ? `${value.lineCap} ${value.dashArray.join(' ')}` : value;
+
+const getCssRawValue = (variableSet: DesignTokenVariableSet, {node, getType}: DesignTokenVariableStructure) => {
+  const nodeType = getType(variableSet, false);
+  if (!nodeType && node.$value) {
+    return typeof node.$value.toString !== undefined ? (node.$value as any).toString() : JSON.stringify(node.$value);
+  }
+  const checkNode = {
+    ...node,
+    $type: node.$type || nodeType
+  } as typeof node;
+
+  // TODO in the following code, `typeof checkNode.$value === 'string' ? checkNode.$value :` is defined to please Jest TS compilation. It should be removed when supported
+  switch (checkNode.$type) {
+    case 'color':
+    case 'number':
+    case 'duration':
+    case 'fontWeight':
+    case 'fontFamily':
+    case 'dimension': {
+      return checkNode.$value.toString();
+    }
+    case 'strokeStyle': {
+      return renderCssTypeStrokeStyleValue(checkNode.$value);
+    }
+    case 'cubicBezier': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        checkNode.$value.join(', ');
+    }
+    case 'border': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        `${checkNode.$value.width} ${renderCssTypeStrokeStyleValue(checkNode.$value.style)} ${checkNode.$value.color}`;
+    }
+    case 'gradient': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        `linear-gradient(0deg, ${checkNode.$value.map(({color, position}) => `${color} ${position}`).join(', ')})`;
```

Comment:
can't we specify the 1st param of linear gradient with design token ?

Reply:
Not yet (cf: https://design-tokens.github.io/community-group/format/#gradient)

Reply:
Maybe we can reference https://github.com/design-tokens/community-group/issues/101
so we can support it when the issue is closed